### PR TITLE
BaSP-MR-2: ADMINS - get, getById and create

### PR DESCRIPTION
**Pull request by Franro212**

### CODE REVIEW

File: src/models/Admin.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    lastName: {
+      type: String,
+      required: true,
+      maxlength: 10,
+    },
+    dni: {
+      type: Number,
+      required: true,
+      min: 7,
```
Comment (AleMac17) [inline src/models/Admin.js:20]: In mongoose min in numbers its not the same as minlength. You has to put the exact max number
`min: 1000000`

**Review by AleMac17 — CHANGES_REQUESTED** [review]

File: src/controllers/admins.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  try {
+    const admins = await Admin.find();
+    return res.status(200).json({
+      message: 'Complete admins list',
+      data: admins,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Error! getting administrators.',
```
Comment (AleMac17) [inline src/controllers/admins.js:13]: In the message of the catch return, we need to put an error.
for example `message: error,`

File: src/controllers/admins.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      data: undefined,
+      error,
```
Comment (AleMac17) [inline src/controllers/admins.js:15]: The: true is missing after the error.
should be: `error: true,` like line 29.

File: src/controllers/admins.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    return res.status(200).send(admin).json({
+      message: 'The administrator was found',
+      data: admin,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Error! getting the administrator.',
+      data: undefined,
+      error,
```
Comment (AleMac17) [inline src/controllers/admins.js:42]: The: true is missing after the error.
should be: `error: true,` like line 29.

File: src/controllers/admins.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    return res.status(200).send(admin).json({
+      message: 'The administrator was found',
+      data: admin,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Error! getting the administrator.',
```
Comment (AleMac17) [inline src/controllers/admins.js:40]: In the message of the catch return, we need to put an error.
for example `message: error,`

File: src/controllers/admins.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    }
+    const createdAdmin = await admin.save();
+    return res.status(201).json({
+      message: 'The administrator was created.',
+      data: createdAdmin,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Error! Failed to create the administrator.',
```
Comment (AleMac17) [inline src/controllers/admins.js:76]: In the message of the catch return, we need to put an error.
for example `message: error,`

File: src/controllers/admins.js [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      data: undefined,
+      error,
```
Comment (AleMac17) [inline src/controllers/admins.js:78]: The: true is missing after the error.
should be: `error: true,` like line 29.

**Review by AleMac17 — CHANGES_REQUESTED** [review]

File: src/controllers/admins.js [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+      firstName,
+      lastName,
+      dni,
+      phone,
+      email,
+      password,
+    });
+
+    if (!admin) {
+      return res.status(404).json({
```
Comment (AleMac17) [inline src/controllers/admins.js:62]: Status 404 is for requests that the requested resource does not exist.
In this case, the object is not valid, the appropriate thing would be to use status 400.

**Review by AleMac17 — CHANGES_REQUESTED** [review]

File: src/controllers/admins.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        message: 'All fields must be completed.',
+        data: {},
+        error: false,
```
Comment (AleMac17) [inline src/controllers/admins.js:65]: Status 404 is considered an error.
should be: `error: true,`

**Review by AleMac17 — CHANGES_REQUESTED** [review]

**Review by AleMac17 — APPROVED** [review]

**Review by AleMac17 — APPROVED** [review]

**Review by nicolasblasco — APPROVED** [review]

**Pull request merged**